### ISSUE TITLE
SC2216: False positive - pipe to cp with source of /dev/stdin, etc

### ISSUE THREAD

**Issue by NeilW** [koalaman/shellcheck#2203]
#### Here's a snippet or screenshot that shows the problem:

```sh

##!/bin/sh
bzcat -f "$1" | zcat -f | cp --sparse=always /dev/stdin "$outfile" && rm -f "$1"

```

#### Here's what shellcheck currently says:
^-- SC2216: Piping to 'cp', a command that doesn't read stdin. Wrong command or missing xargs?

#### Here's what I wanted or expected to see:

Should pass cleanly if the source device cp is accessing is the correct character device for stdin via one of the myriad of symlinks to it.
